code/modules/mob/living/simple_animal/friendly/dog.dm

```diff
@@ -359,6 +359,10 @@
 					if(isturf(S.loc) || ishuman(S.loc))
 						movement_target = S
 						break
+				for(var/obj/item/toy/rubber_ball/B in oview(src, 4))
```

Comment:
Don't wanna do work on this in case I make a mistake. How do you want it implemented?